test/unit/ParaTest/Console/Commands/ParaTestCommandTest.php

```diff
@@ -47,7 +47,9 @@ public function testConfiguredDefinitionWithPHPUnitTester()
             new InputOption('coverage-clover', null, InputOption::VALUE_REQUIRED, 'Generate code coverage report in Clover XML format.'),
             new InputOption('coverage-html', null, InputOption::VALUE_REQUIRED, 'Generate code coverage report in HTML format.'),
             new InputOption('coverage-php', null, InputOption::VALUE_REQUIRED, 'Serialize PHP_CodeCoverage object to file.'),
-            new InputOption('testsuite', null, InputOption::VALUE_OPTIONAL, 'Filter which testsuite to run')
+            new InputOption('testsuite', null, InputOption::VALUE_OPTIONAL, 'Filter which testsuite to run'),
+            new InputOption('max-batch-size', 'm', InputOption::VALUE_REQUIRED, 'Max batch size (only for functional mode).', 50),
```

Comment:
One test method can have more then one test case. So, actually we have three modes: parallel by test file, parallel by test method, and parallel by test case. My implementation replaces test method parallel with test case parallel. Even if we specify max-batch-size=1 new code will not work the same as old for methods with @dataProvider. I don't know if we actually need all of that methods =) Parallel by test case works the most effective for my needs. Also I found that testsuite grouping can be refactored to be activated on phase when engine group tests in batches - it will reduce amount of code. Current implementation has different logics (and different classes) for two modes: functional and regular.